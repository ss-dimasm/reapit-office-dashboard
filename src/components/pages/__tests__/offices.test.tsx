import React from 'react'
import { mount } from 'enzyme'
import { QueryClient, QueryClientProvider } from 'react-query'

import { Card } from '@reapit/elements'

import Offices from '../offices'
import LoadingIndicator from '../../ui/loading'
import { useGetOfficesQuery } from '../../../generated/graphql'
import { OFFICE_PAGED_EXAMPLE } from '../__mocks__/office.mock'

const mockedUseGetOfficeQuery = useGetOfficesQuery as jest.Mock<any>

jest.mock('../../../generated/graphql')

const queryClient = new QueryClient()

describe('Office Page', () => {
  beforeEach(() => {
    mockedUseGetOfficeQuery.mockImplementation(() => ({
      isLoading: true,
      data: OFFICE_PAGED_EXAMPLE,
    }))
  })

  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render Office component', () => {
    const wrapper = mount(
      <QueryClientProvider client={queryClient}>
        <Offices />
      </QueryClientProvider>
    )

    expect(wrapper).toMatchSnapshot()
  })
  it('should render loading indicator first', () => {
    mockedUseGetOfficeQuery.mockImplementation(() => ({
      isLoading: true,
      data: {
        OfficeData: OFFICE_PAGED_EXAMPLE,
      },
    }))

    const wrapper = mount(
      <QueryClientProvider client={queryClient}>
        <Offices />
      </QueryClientProvider>
    )

    expect(wrapper.find(LoadingIndicator)).toHaveLength(1)
    expect(wrapper.find(LoadingIndicator).text()).not.toBeNull()
    expect(wrapper.find(LoadingIndicator).debug()).toMatchSnapshot()
  })

  it('should render office data', () => {
    mockedUseGetOfficeQuery.mockImplementation(() => ({
      isLoading: false,
      data: {
        OfficeData: OFFICE_PAGED_EXAMPLE,
      },
    }))

    const wrapper = mount(
      <QueryClientProvider client={queryClient}>
        <Offices />
      </QueryClientProvider>
    )

    const card = wrapper.find(Card)
    expect(card).toHaveLength(1)
    expect(card.text().includes('Reapit')).toBeTruthy()
    expect(card).toMatchSnapshot()
  })
})
