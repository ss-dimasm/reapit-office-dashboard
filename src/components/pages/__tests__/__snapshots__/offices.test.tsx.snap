// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Office Page should render Office component 1`] = `
<QueryClientProvider
  client={
    QueryClient {
      "defaultOptions": Object {},
      "mutationCache": MutationCache {
        "config": Object {},
        "listeners": Array [],
        "mutationId": 0,
        "mutations": Array [],
      },
      "mutationDefaults": Array [],
      "queryCache": QueryCache {
        "config": Object {},
        "listeners": Array [],
        "queries": Array [],
        "queriesMap": Object {},
      },
      "queryDefaults": Array [],
      "unsubscribeFocus": [Function],
      "unsubscribeOnline": [Function],
    }
  }
>
  <Offices>
    <LoadingIndicator
      isFullScreen={true}
      label="Please wait..."
    >
      <div
        className="el-flex el-flex-justify-center elH-full el-flex-align-center"
        style={
          Object {
            "height": "70%",
          }
        }
      >
        <Ut
          data-testid="loadingIndicator"
          label="Please wait..."
        >
          <ElLoaderContainer
            className=""
          >
            <div>
              <ElLoaderLabel>
                <p>
                  Please wait...
                </p>
              </ElLoaderLabel>
              <ElLoader
                data-testid="loadingIndicator"
              >
                <div
                  data-testid="loadingIndicator"
                >
                  <ElLoaderMovingBar>
                    <div />
                  </ElLoaderMovingBar>
                </div>
              </ElLoader>
            </div>
          </ElLoaderContainer>
        </Ut>
      </div>
    </LoadingIndicator>
  </Offices>
</QueryClientProvider>
`;

exports[`Office Page should render loading indicator first 1`] = `
"<LoadingIndicator label=\\"Please wait...\\" isFullScreen={true}>
  <div className=\\"el-flex el-flex-justify-center elH-full el-flex-align-center\\" style={{...}}>
    <Ut data-testid=\\"loadingIndicator\\" label=\\"Please wait...\\">
      <ElLoaderContainer className=\\"\\">
        <div className={[undefined]}>
          <ElLoaderLabel>
            <p className={[undefined]}>
              Please wait...
            </p>
          </ElLoaderLabel>
          <ElLoader data-testid=\\"loadingIndicator\\">
            <div data-testid=\\"loadingIndicator\\" className={[undefined]}>
              <ElLoaderMovingBar>
                <div className={[undefined]} />
              </ElLoaderMovingBar>
            </div>
          </ElLoader>
        </div>
      </ElLoaderContainer>
    </Ut>
  </div>
</LoadingIndicator>"
`;

exports[`Office Page should render office data 1`] = `
<Component
  data-testid="office-card"
  hasListCard={true}
  listCardHeading="Reapit"
  listCardItems={
    Array [
      Object {
        "listCardItemHeading": "Manager",
        "listCardItemIcon": "applicantInfographic",
        "listCardItemSubHeading": "Mr John Smith",
      },
      Object {
        "listCardItemHeading": "Email",
        "listCardItemIcon": "emailSystem",
        "listCardItemSubHeading": "example@email.com",
      },
    ]
  }
  listCardSubHeading="Join since 2018-12-12T12:30:23.0000000Z"
  onClick={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "margin": "0.5rem",
      "width": "32%",
    }
  }
>
  <Ke
    data-testid="office-card"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "margin": "0.5rem",
        "width": "32%",
      }
    }
  >
    <ElCardWrap
      data-testid="office-card"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "margin": "0.5rem",
          "width": "32%",
        }
      }
    >
      <div
        data-testid="office-card"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "margin": "0.5rem",
            "width": "32%",
          }
        }
      >
        <rt>
          <ElCardListMainWrap>
            <div>
              <Ct>
                <ElCardListHeading>
                  <h5>
                    Reapit
                  </h5>
                </ElCardListHeading>
              </Ct>
              <nt>
                <ElCardListSubHeading>
                  <h6>
                    Join since 2018-12-12T12:30:23.0000000Z
                  </h6>
                </ElCardListSubHeading>
              </nt>
              <at
                className="el-mobile-list-toggle"
                onClick={[Function]}
              >
                <ElMobileToggle
                  className="el-mobile-list-toggle"
                  onClick={[Function]}
                >
                  <button
                    onClick={[Function]}
                  >
                    <Ze
                      icon="arrowDownSystem"
                    >
                      <ElIcon
                        style={
                          Object {
                            "fontSize": undefined,
                          }
                        }
                      >
                        <span
                          style={
                            Object {
                              "fontSize": undefined,
                            }
                          }
                        >
                          <Memo()>
                            <svg
                              fill="none"
                              height="1em"
                              viewBox="0 0 12 12"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clipRule="evenodd"
                                d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                                fill="currentColor"
                                fillRule="evenodd"
                              />
                            </svg>
                          </Memo()>
                        </span>
                      </ElIcon>
                    </Ze>
                  </button>
                </ElMobileToggle>
              </at>
            </div>
          </ElCardListMainWrap>
        </rt>
        <dt
          key="0"
        >
          <ElCardListItem>
            <div>
              <ot>
                <ElCardListIcon>
                  <div>
                    <Ze
                      icon="applicantInfographic"
                    >
                      <ElIcon
                        style={
                          Object {
                            "fontSize": undefined,
                          }
                        }
                      >
                        <span
                          style={
                            Object {
                              "fontSize": undefined,
                            }
                          }
                        >
                          <Memo()>
                            <svg
                              fill="none"
                              height="1em"
                              viewBox="0 0 20 20"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <mask
                                height="1em"
                                id="mask0"
                                mask-type="alpha"
                                maskUnits="userSpaceOnUse"
                                width="1em"
                                x="0"
                                y="0"
                              >
                                <circle
                                  cx="10"
                                  cy="10"
                                  fill="#F2F2F2"
                                  r="10"
                                />
                              </mask>
                              <g
                                mask="url(#mask0)"
                              >
                                <rect
                                  fill="#F2F2F2"
                                  height="1em"
                                  width="1em"
                                />
                                <path
                                  d="M15.4305 15.6062V16.9906C15.4305 17.1465 15.3042 17.2727 15.1484 17.2727H4.8518C4.69594 17.2727 4.5697 17.1465 4.5697 16.9906V15.6062C4.5697 14.1538 5.74711 12.9771 7.19991 12.9767C7.71333 12.9764 8.16539 12.7122 8.42562 12.3117C8.68706 11.9099 8.66012 11.5605 8.66012 11.2781L8.66541 11.2651C7.39949 10.7411 6.50912 9.49386 6.50912 8.03858V7.92821C6.00646 7.92821 5.62757 7.53098 5.62757 7.06288V5.59738C5.61628 5.08714 5.63956 3.28771 6.93932 2.03484C8.45066 0.57816 10.8457 0.697346 12.5806 1.47664C12.637 1.50203 12.6352 1.58278 12.5781 1.60641L12.1257 1.79259C12.058 1.82045 12.0724 1.92024 12.1451 1.928L12.3648 1.95162C13.4166 2.06411 14.3938 2.63536 14.9996 3.58603C15.0158 3.61177 15.0112 3.65091 14.9904 3.67312C14.6166 4.06841 14.4079 4.59311 14.4079 5.13862V7.06288C14.4079 7.54068 14.0203 7.92821 13.5425 7.92821H13.491V8.03858C13.491 9.49386 12.6007 10.7411 11.3348 11.2651L11.34 11.2781C11.34 11.5524 11.3109 11.9065 11.5745 12.3117C11.8348 12.7122 12.2868 12.9764 12.8003 12.9767C14.2531 12.9771 15.4305 14.1538 15.4305 15.6062Z"
                                  fill="#FFE5C7"
                                />
                                <path
                                  d="M15.4305 15.6063V19.718C15.4305 19.8738 15.3042 20.0001 15.1484 20.0001H4.8518C4.69601 20.0001 4.5697 19.8738 4.5697 19.718V15.6063C4.5697 14.1539 5.74728 12.9773 7.19973 12.9768C7.71329 12.9766 8.16546 12.7122 8.42562 12.3118C9.33377 13.0876 10.6659 13.0881 11.5745 12.3118C11.8347 12.7122 12.2869 12.9766 12.8004 12.9768C14.2529 12.9773 15.4305 14.1539 15.4305 15.6063Z"
                                  fill="#7A2C81"
                                />
                                <path
                                  d="M14.9994 3.5863C15.0157 3.61201 15.0112 3.65094 14.9902 3.67308C14.6166 4.06866 14.4078 4.59318 14.4078 5.13872V7.06298C14.4078 7.49981 14.0787 7.92832 13.491 7.92832V5.75017C13.491 4.81537 12.7332 4.05758 11.7984 4.05758H8.20168C7.26687 4.05758 6.50909 4.81537 6.50909 5.75017V7.92832C6.00643 7.92832 5.62754 7.53109 5.62754 7.06298V5.59748C5.61625 5.08724 5.63952 3.28781 6.93929 2.03494C8.45052 0.578227 10.8458 0.697483 12.5804 1.47685C12.6368 1.5022 12.6353 1.58299 12.5781 1.60654L12.1257 1.7928C12.058 1.82066 12.0723 1.92034 12.145 1.92813L12.3647 1.95172C13.4168 2.06411 14.3936 2.63535 14.9994 3.5863Z"
                                  fill="#42434D"
                                />
                                <path
                                  d="M13.4909 8.0386C13.4909 9.9673 11.9271 11.5296 9.99999 11.5296C8.18427 11.5296 6.50903 10.118 6.50903 7.92823V5.75008C6.50903 4.81528 7.26682 4.0575 8.20162 4.0575H11.7984C12.7332 4.0575 13.4909 4.81528 13.4909 5.75008V8.0386Z"
                                  fill="#FACD8E"
                                />
                                <path
                                  d="M12.8563 16.6733H12.01C11.8542 16.6733 11.7279 16.547 11.7279 16.3912C11.7279 16.2354 11.8542 16.1091 12.01 16.1091H12.8563C13.0121 16.1091 13.1384 16.2354 13.1384 16.3912C13.1384 16.547 13.0121 16.6733 12.8563 16.6733Z"
                                  fill="white"
                                />
                                <path
                                  d="M9.20955 10.293C9.78352 10.6521 9.33092 11.5327 8.70262 11.2805C7.41083 10.762 6.50903 9.50314 6.50903 8.03877V5.75025C6.50903 5.39467 6.6187 5.06469 6.80601 4.79225C7.03571 4.45821 7.5669 4.60578 7.5669 5.04501V7.33353C7.5669 8.62095 8.24365 9.68869 9.20955 10.293Z"
                                  fill="#FFE5C7"
                                />
                                <path
                                  d="M11.3799 11.8572C10.5111 12.2208 9.55638 12.2491 8.62024 11.8572C8.66805 11.6588 8.66009 11.5303 8.66009 11.2782L8.66537 11.2651C9.51456 11.6176 10.4839 11.6183 11.3347 11.2651L11.34 11.2782C11.34 11.5304 11.332 11.6589 11.3799 11.8572Z"
                                  fill="#F5B555"
                                />
                                <path
                                  d="M8.6207 13.8286C8.4903 13.8697 8.35045 13.8923 8.20051 13.8936C6.74432 13.9064 5.55704 15.0578 5.55704 16.5141V16.7791C5.55704 17.0518 5.33602 17.2728 5.06337 17.2728H4.8518C4.69601 17.2728 4.5697 17.1465 4.5697 16.9907V15.6063C4.5697 14.1538 5.74728 12.9773 7.19973 12.9768C7.71329 12.9766 8.16546 12.7122 8.42562 12.3118C8.53049 12.4013 8.64277 12.4818 8.76104 12.5523C9.28126 12.8623 9.19836 13.647 8.6207 13.8286Z"
                                  fill="#A245AA"
                                />
                                <path
                                  d="M11.3795 13.8286C11.5099 13.8697 11.6498 13.8923 11.7997 13.8936C13.2559 13.9064 14.4432 15.0578 14.4432 16.5141V16.7791C14.4432 17.0518 14.6642 17.2728 14.9369 17.2728H15.1484C15.3042 17.2728 15.4305 17.1465 15.4305 16.9907V15.6063C15.4305 14.1538 14.253 12.9773 12.8005 12.9768C12.2869 12.9766 11.8348 12.7122 11.5746 12.3118C11.4697 12.4013 11.3575 12.4818 11.2392 12.5523C10.719 12.8623 10.8019 13.647 11.3795 13.8286Z"
                                  fill="#63186A"
                                />
                                <path
                                  d="M12.5781 1.60641C12.1635 1.77704 12.1044 1.81466 11.9607 1.79238C9.64489 1.43341 7.61308 2.39159 6.92307 4.64107V4.64142C6.66495 4.93833 6.50909 5.32622 6.50909 5.75007V7.07321C6.50909 7.64404 5.62754 7.68378 5.62754 7.06288V5.59738C5.61625 5.08714 5.63952 3.28771 6.93929 2.03484C8.45063 0.57816 10.8456 0.697346 12.5805 1.47664C12.637 1.50203 12.6352 1.58278 12.5781 1.60641Z"
                                  fill="#4D4E59"
                                />
                              </g>
                            </svg>
                          </Memo()>
                        </span>
                      </ElIcon>
                    </Ze>
                  </div>
                </ElCardListIcon>
              </ot>
              <it>
                <ElCardListItemTextWrap>
                  <div>
                    <st>
                      <ElCardListItemTextPrimary>
                        <div>
                          Manager
                        </div>
                      </ElCardListItemTextPrimary>
                    </st>
                    <ct>
                      <ElCardListItemTextSecondary>
                        <div>
                          Mr John Smith
                        </div>
                      </ElCardListItemTextSecondary>
                    </ct>
                  </div>
                </ElCardListItemTextWrap>
              </it>
            </div>
          </ElCardListItem>
        </dt>
        <dt
          key="1"
        >
          <ElCardListItem>
            <div>
              <ot>
                <ElCardListIcon>
                  <div>
                    <Ze
                      icon="emailSystem"
                    >
                      <ElIcon
                        style={
                          Object {
                            "fontSize": undefined,
                          }
                        }
                      >
                        <span
                          style={
                            Object {
                              "fontSize": undefined,
                            }
                          }
                        >
                          <Memo()>
                            <svg
                              fill="none"
                              height="1em"
                              viewBox="0 0 20 20"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g>
                                <path
                                  d="M18.125 2.5H1.87501C0.841067 2.5 0 3.34107 0 4.37501V15.625C0 16.659 0.841067 17.5 1.87501 17.5H18.125C19.1589 17.5 20 16.659 20 15.625V4.37501C20 3.34107 19.1589 2.5 18.125 2.5ZM18.125 3.74999C18.2099 3.74999 18.2907 3.76761 18.3645 3.79838L10 11.048L1.63542 3.79838C1.70927 3.76765 1.79005 3.74999 1.87497 3.74999H18.125ZM18.125 16.25H1.87501C1.53017 16.25 1.24999 15.9698 1.24999 15.625V5.11901L9.59045 12.3474C9.70826 12.4493 9.85413 12.5 10 12.5C10.1459 12.5 10.2917 12.4494 10.4096 12.3474L18.75 5.11901V15.625C18.75 15.9698 18.4698 16.25 18.125 16.25Z"
                                  fill="currentColor"
                                />
                              </g>
                              <defs>
                                <clipPath
                                  id="clip0"
                                >
                                  <rect
                                    fill="white"
                                    height="1em"
                                    width="1em"
                                  />
                                </clipPath>
                              </defs>
                            </svg>
                          </Memo()>
                        </span>
                      </ElIcon>
                    </Ze>
                  </div>
                </ElCardListIcon>
              </ot>
              <it>
                <ElCardListItemTextWrap>
                  <div>
                    <st>
                      <ElCardListItemTextPrimary>
                        <div>
                          Email
                        </div>
                      </ElCardListItemTextPrimary>
                    </st>
                    <ct>
                      <ElCardListItemTextSecondary>
                        <div>
                          example@email.com
                        </div>
                      </ElCardListItemTextSecondary>
                    </ct>
                  </div>
                </ElCardListItemTextWrap>
              </it>
            </div>
          </ElCardListItem>
        </dt>
      </div>
    </ElCardWrap>
  </Ke>
</Component>
`;
